src/mon/OSDMonitor.cc

```diff
@@ -7510,6 +7510,10 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
                 << "' device_class '" << device_class << "'"
                 << dendl;
         err = newcrush.update_device_class(osd, device_class, name, &ss);
+        if (err == 0) {
```

Comment:
@guzhongyan after a second thought, i think we should add the ". " in `update_device_class()`,  as it is difficult to figure out if that method has appended any error message to `ss` or not by checking its return value. and what i was looking at https://github.com/ceph/ceph/pull/21174/files#diff-0a5db46a44ae9900e226289a810f10e8R7524, where we need to add `". "` also.

sorry for the bikeshedding. thought it was simpler, but we can hardly cover all cases in that way.

Reply:
@tchaikov , updated, please take a look.